### ISSUE TITLE
Refactoring

### ISSUE THREAD

**Issue by obiwan87** [obiwan87/odin-intellij#40]
Add refactoring handlers for:

- [ ] Introduce variable 
- [ ] Extract method
- [ ] Safe delete
- [ ] Remove unused imports
- [ ] Rename 
- [ ] Inline renaming for identifiers with known scope
- [ ] Extract parameter (Ctrl+Alt+P)
- [ ] to be continued

**Comment by Dima-369** [comment]
How about `Specify type explicitly`? I don't need it often, but when I need it, it's really useful in other languages as well.

For instance, here in TypeScript when I am not fully sure of the type, and the type could be obfuscated from generics.

![CleanShot 2024-08-13 at 14 19 01@2x](https://github.com/user-attachments/assets/6f578d73-a166-4d1e-bcf6-52f1cd39416a)

---

Like for this trivial example:

```odin
points_arr := make([]f32, 3 * n)
```

**Comment by obiwan87** [comment]
> How about `Specify type explicitly`? I don't need it often, but when I need it, it's really useful in other languages as well.
Added it to the list ;-)

**Comment by obiwan87** [comment]
@Dima-369 Added support for "Specify type explicitly". I already found a little bug. The intention is not triggered at the end of an identifier, only at the beginning and in the middle of it

```
identifier := value        
```
so if the cursor is after the "r" it won't trigger

**Comment by Dima-369** [comment]
> Added support for "Specify type explicitly"

Cool, I tested with `fmt.println()` and with a local proc in the same file, and it worked flawlessly! This is really useful, thanks!

But, it fails on `os.args`, probably because it is defined like this.

```odin
// "Argv" arguments converted to Odin strings
args := _alloc_command_line_arguments()
```

![CleanShot 2024-08-21 at 15 05 53@2x](https://github.com/user-attachments/assets/5c712772-43d3-417c-bfbd-66fdebe4596c)

---

It also fails on this: `lines := strings.split("foo", "\n")`

**Comment by obiwan87** [comment]
Yes, procedures with multiple values are not supported yet. Would it be always safe to just infer the type of the first return argument?

**Comment by Dima-369** [comment]
> how Odin handles multiple return values, when stored in only one variable. Would it be always safe to just infer the type of the first return argument?

I think so, yes.

**Comment by obiwan87** [comment]
Got ya. 

There is so much more to this functionality. I thought about this feature this morning and there is a lot more to it that I haven't yet implemented. With composite types like procedures types maps and instantiated poly structs it should not work either. 

Basically you have to analyze all the types a type might contain and import the package they're in, or use the import that is already defined in the current file.
